### PR TITLE
Add commands Ltac2 Globalize, Ltac2 Check, Print Ltac2 Type + related fixes

### PR DESCRIPTION
**Pull request by SkySkimmer**
Overlays:
- https://github.com/SkySkimmer/coq-ltac2-compiler/pull/3

### CODE REVIEW

**Comment by SkySkimmer** [comment]
I don't have a deep understanding of printing levels so I may have gotten them wrong.

**Comment by SkySkimmer** [comment]
The goal of Ltac2 Globalize is meant to help with errors involving notations, but it's pretty limited as eg the example from https://github.com/coq/coq/issues/17869 fails to globalize since the notation contains ctacext (for constr).

**Comment by SkySkimmer** [comment]
>The goal of Ltac2 Globalize is meant to help with errors involving notations, but it's pretty limited as eg the example from #17869 fails to globalize since the notation contains ctacext (for constr).

Actually we can make it work, cf last commit, although it feels a bit hacky

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: doc/sphinx/proof-engine/ltac2.rst [hunk 1]
```diff
@@ -350,6 +350,14 @@ Printing Ltac2 tactics
    :cmd:`Print` can print defined Ltac2 tactics and can avoid printing
    other objects by using `Print Ltac2`.
 
+.. cmd:: Ltac2 Globalize @ltac2_expr
+
+   This command prints the result of resolving notations in the given expression.
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:355]:
```suggestion
   Prints the result of resolving notations in the given expression.
```

We usually omit boilerplate phrases that would appear in hundreds of places such as "This command".  The doc already shows "Command" next to the syntax.

File: doc/sphinx/proof-engine/ltac2.rst [hunk 2]
```diff
@@ -350,6 +350,14 @@ Printing Ltac2 tactics
...
+
+.. cmd:: Ltac2 Check @ltac2_expr
+
+   This command typechecks the given expression and prints the result.
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/ltac2.rst:359]:
```suggestion
   Typechecks the given expression and prints the result.
```

**Review by jfehrle — COMMENTED** [review]
wording suggestions

Also, don't you need some OCaml code that implements `Ltac2 Globalize` and `Ltac2 Check`?

**Comment by SkySkimmer** [comment]
>Also, don't you need some OCaml code that implements Ltac2 Globalize and Ltac2 Check?

There is ocaml code though. Are you looking at the diff with some file filter or something?

**Comment by ejgallego** [comment]
The Globalize stuff is cool, there are a couple of CEP and a few issues in SerAPI / coq-lsp to implement this for all commands (both for notations and for name resolution) I'd be nice to do so but that's a lot of work on the other hand.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**